core/src/main/java/org/apache/calcite/rex/RexSimplify.java

```diff
@@ -2710,6 +2710,16 @@ private boolean accept1(RexNode e, SqlKind kind,
       }
     }
 
+    /** Checks whether it is worth to fix and convert to {@code SEARCH} calls. */
```

Comment:
s/Checks/Returns/; "Checks" often means that the method will throw if the check fails, whereas "Returns" is unambiguous.

Put the criteria in the javadoc, and describe the purpose of `newTermsCnt`.

I wouldn't abbreviate `count` to `cnt`. Save to characters, but convert a word into a non-word.